src/knex.js

```diff
@@ -0,0 +1,83 @@
+import Raw from './raw';
+import Client from './client';
+
+import makeKnex from './util/make-knex';
+import parseConnection from './util/parse-connection';
+import { SUPPORTED_CLIENTS, CLIENT_ALIASES } from './constants';
+
+export default function Knex(config) {
+  // If config is string, try to parse it
+  if (typeof config === 'string') {
+    const parsedConfig = Object.assign(parseConnection(config), arguments[2]);
```

Comment:
_.assign replaced with Object.assign in this file since it is supported since Node 4.